datasets/squad_kor_v2/squad_kor_v2.py

```diff
@@ -0,0 +1,149 @@
+"""
+_HOMEPAGE = "https://korquad.github.io/"
+_LICENSE = "CC BY-ND 2.0 KR"
+
+_URL = "https://github.com/korquad/korquad.github.io/raw/master/dataset/KorQuAD_2.1"
+_URLS = {
+    "train": [
+        _URL + "/train/KorQuAD_2.1_train_00.zip",
+        _URL + "/train/KorQuAD_2.1_train_01.zip",
+        _URL + "/train/KorQuAD_2.1_train_02.zip",
+        _URL + "/train/KorQuAD_2.1_train_03.zip",
+        _URL + "/train/KorQuAD_2.1_train_04.zip",
+        _URL + "/train/KorQuAD_2.1_train_05.zip",
+        _URL + "/train/KorQuAD_2.1_train_06.zip",
+        _URL + "/train/KorQuAD_2.1_train_07.zip",
+        _URL + "/train/KorQuAD_2.1_train_08.zip",
+        _URL + "/train/KorQuAD_2.1_train_09.zip",
+        _URL + "/train/KorQuAD_2.1_train_10.zip",
+        _URL + "/train/KorQuAD_2.1_train_11.zip",
+        _URL + "/train/KorQuAD_2.1_train_12.zip",
+    ],
+    "validation": [_URL + "/dev/KorQuAD_2.1_dev_00.zip", _URL + "/dev/KorQuAD_2.1_dev_01.zip"],
+}
+
+
+class SquadKorV2(datasets.GeneratorBasedBuilder):
+    """KorQuAD 2.1 dataset"""
+
+    VERSION = datasets.Version("2.1.0")
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="squad_kor_v2", version=VERSION, description=_DESCRIPTION),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "title": datasets.Value("string"),
+                    "context": datasets.Value("string"),
+                    "question": datasets.Value("string"),
+                    "answer": datasets.Features(
+                        {
+                            "text": datasets.Value("string"),
+                            "answer_start": datasets.Value("int32"),
+                            "html_answer_start": datasets.Value("int32"),
+                        }
+                    ),
+                    "url": datasets.Value("string"),
+                    "raw_html": datasets.Value("string"),
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # download and extract URLs
+        urls_to_download = _URLS
+        downloaded_files = dl_manager.download_and_extract(urls_to_download)
+
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"dirs": downloaded_files["train"]}),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION, gen_kwargs={"dirs": downloaded_files["validation"]}
+            ),
+        ]
+
+    def _generate_examples(self, dirs):
+        """Yields examples."""
+
+        for d in dirs:
+            filepaths = os.scandir(d)
+            for filepath in filepaths:
```

Comment:
```suggestion
            for filepath in sorted(filepaths):
```
we need the order here to be deterministic